lambda-code/common/dbConnects.js

```
@@ -0,0 +1,95 @@
+const sql = require('mssql');
+const AWS = require('aws-sdk');
+const https = require('https');
+const { PromisePool } = require('@supercharge/promise-pool');
+
+let usePool = true;
+const dynamoClient = new AWS.DynamoDB({
+    apiVersion: '2012-08-10',
+    region: "eu-west-1",
+    accessKeyId: null,
+    secretAccessKey: null,
+    endpoint: null,
+    httpOptions: {
+        agent: new https.Agent({keepAlive: true })
+    }
+});
+
+function dynamoGetProductInfo(dynamo, erigthsID){
+    const id = erigthsID.toString();
+    const query = {
+        TableName: 'OUP_ERIGHTS_ID_MAPPING_dev-primary',
+        ProjectionExpression: "OUP_ID, ERIGHTS_ID",
+        IndexName: "ERIGHTS_ID-ENTITY-index",
+        KeyConditionExpression: "ERIGHTS_ID = :id",
+        ConsistentRead: false,
+        ExpressionAttributeValues: {
+            ":id": {"S": id}
+        }
+    };
+    return dynamo.query(query).promise();
+}
+
+let mappingFail = [];
+
+async function dynamoMappedData(erightsIDs = [], pageSize = 10){
+  if(usePool){
+    const { results } = await PromisePool
+      .withConcurrency(pageSize)
+      .for(erightsIDs)
+      .process(async erightsId => {
+        const res = await dynamoGetProductInfo(dynamoClient, erightsId);
+        if(res.Items.length === 0){
+          mappingFail.push(erightsId);
+        }
+        return res;
+      });
+    return results;
+  } else {
+    var i,j;
+    let items = [];
+    for (i = 0,j = erightsIDs.length; i < j; i += pageSize) {
+      const page = erightsIDs.slice(i, i + pageSize);        
+      const results = await Promise.all(page.map(async erightsId => {
+        const res = await dynamoGetProductInfo(dynamoClient, erightsId);
+        if(res.Items.length === 0){
+          console.log(`FAILED TO MAP ${erightsId}`);
```

Forgot about this. It should add the erightsId to the list of failed mappings, instead of simply logging it